fog/sample-paykit/src/client_builder.rs

```diff
@@ -177,42 +205,108 @@ impl ClientBuilder {
         FogKeyImageGrpcClient,
         FogUntrustedLedgerGrpcClient,
     ) {
-        let verifier = {
-            let mr_signer_verifier =
-                mc_fog_ledger_enclave_measurement::get_mr_signer_verifier(None);
-
-            let mut verifier = Verifier::default();
-            verifier.mr_signer(mr_signer_verifier).debug(DEBUG_ENCLAVE);
-            verifier
-        };
+        let verifier = self.get_fog_ledger_verifier();
 
         log::debug!(
             self.logger,
             "Fog ledger attestation verifier: {:?}",
             verifier
         );
 
-        let client_uri = FogLedgerUri::from_str(&self.ledger_server_address).unwrap_or_else(|e| {
-            panic!(
-                "Could not parse client uri: {}: {:?}",
-                self.ledger_server_address, e
-            )
-        });
-
         (
             FogMerkleProofGrpcClient::new(
-                client_uri.clone(),
+                self.ledger_server_address.clone(),
                 verifier.clone(),
                 grpc_env.clone(),
                 self.logger.clone(),
             ),
             FogKeyImageGrpcClient::new(
-                client_uri.clone(),
+                self.ledger_server_address.clone(),
                 verifier,
                 grpc_env.clone(),
                 self.logger.clone(),
             ),
-            FogUntrustedLedgerGrpcClient::new(client_uri, grpc_env, self.logger.clone()),
+            FogUntrustedLedgerGrpcClient::new(
+                self.ledger_server_address.clone(),
+                grpc_env,
+                self.logger.clone(),
+            ),
         )
     }
+
+    // Get consensus verifier (dynamic or build time, MRSIGNER)
+    fn get_consensus_verifier(&self) -> Verifier {
+        let mr_signer_verifier = if let Some(signature) = self.consensus_sigstruct.as_ref() {
+            let mut mr_signer_verifier = MrSignerVerifier::new(
+                signature.mrsigner().into(),
+                signature.product_id(),
+                signature.version(),
+            );
+            mr_signer_verifier.allow_hardening_advisories(&["INTEL-SA-00334"]);
+            mr_signer_verifier
+        } else {
+            mc_consensus_enclave_measurement::get_mr_signer_verifier(None)
+        };
+
+        let mut verifier = Verifier::default();
+        verifier.debug(DEBUG_ENCLAVE).mr_signer(mr_signer_verifier);
+        verifier
+    }
+
+    // Get fog ingest verifier (dynamic or build time, MRSIGNER)
+    fn get_fog_ingest_verifier(&self) -> Verifier {
+        let mr_signer_verifier = if let Some(signature) = self.fog_ingest_sigstruct.as_ref() {
+            let mut mr_signer_verifier = MrSignerVerifier::new(
+                signature.mrsigner().into(),
+                signature.product_id(),
+                signature.version(),
+            );
+            mr_signer_verifier.allow_hardening_advisories(&["INTEL-SA-00334"]);
```

Comment:
Would it make sense to move this string to be a const somewhere that is referred to multiple times to avoid missing one in the event that we need to change this?

Reply:
Yea, I'm trying to follow the patterns established elsewhere , but it would be nice if there were a crate that owned the "standard" configuration of attest verifiers so that we don't copy paste these strings around